elide-contrib/elide-test-helpers/src/main/java/com/yahoo/elide/contrib/testhelpers/graphql/GraphQLDSL.java

```diff
@@ -0,0 +1,482 @@
+import com.yahoo.elide.contrib.testhelpers.graphql.elements.VariableDefinitions;
+import com.yahoo.elide.contrib.testhelpers.graphql.elements.Arguments;
+import com.yahoo.elide.contrib.testhelpers.graphql.elements.Edges;
+import com.yahoo.elide.contrib.testhelpers.graphql.elements.Node;
+import com.yahoo.elide.contrib.testhelpers.graphql.elements.Selection;
+import com.yahoo.elide.contrib.testhelpers.graphql.elements.SelectionSet;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * {@link GraphQLDSL} programmatically construct GraphQL query(excluding query variables) and serializes it to a GraphQL
+ * query string.
+ * <p>
+ * For example,
+ * <pre>
+ * {@code
+ * String graphQLQuery = document(
+ *         typedOperation(
+ *                 TypedOperation.OperationType.QUERY,
+ *                 "myQuery",
+ *                 variableDefinitions(
+ *                         variableDefinition("bookId", "[String]")
+ *                 ),
+ *                 selections(
+ *                         entity(
+ *                                 "book",
+ *                                 arguments(
+ *                                         argument("ids", variableValue("bookId"))
+ *                                 ),
+ *                                 selections(
+ *                                         field("id"),
+ *                                         field("title"),
+ *                                         field(
+ *                                                 "authors",
+ *                                                 selection(
+ *                                                         field("name")
+ *                                                 )
+ *                                         )
+ *                                 )
+ *                         )
+ *                 )
+ *         )
+ * ).toQuery();
+ * }
+ * </pre>
+ * will produces the following query
+ * <pre>
+ * {@code
+ * query myQuery($bookId: [String]) {
+ *     book(ids: $bookId) {
+ *         edges {
+ *             node {
+ *                 id
+ *                 title
+ *                 authors {
+ *                     edges {
+ *                         node {
+ *                             name
+ *                         }
+ *                     }
+ *                 }
+ *             }
+ *         }
+ *     }
+ * }
+ * }
+ * </pre>
+ * In addition, {@link GraphQLDSL} also allows programmatically constructing GraphQL response data. For example,
+ * <pre>
+ * {@code
+ * document(
+ *         selection(
+ *                 responseField(
```

Comment:
The plan was to combine responseField and field into a single concept.  THe comments need to all be updated to reflect the DSL changes.